### PR TITLE
Submit rating button adds/edits a rating

### LINKED ISSUE #210: Submitting a rating should add rating to database if user has not rated before
~~Blocked by #212~~

### LINKED ISSUE #211: Submitting on the submit rating screen should edit an existing rating if the user has rated the place before
Blocked by #212

### LINKED ISSUE #256: Change rating submission on submit rating screen to match new rating design
Update the names of the labels on the submit rating page to match our new design mockup

### PR DESCRIPTION
**Pull request by carpenterd777**
Closes #210, closes #211, closes #256

### CODE REVIEW

File: src/components/PopUp.tsx [hunk 1]
```diff
@@ -50,6 +66,8 @@ export const PopUp: React.FC<PopUpProps> = ({
 					/>
 				</View>
 			</Modal>
+			{/* {renderButton()} */}
```
Comment (aej11a) [inline src/components/PopUp.tsx:69]: pls delete

**Review by elenirotsides — CHANGES_REQUESTED** [review]
I am unable to load anything past the main page. `peer-types` has changed a bit from 1.0.2 to 1.0.4 and I think that may be why I can't get the details page to load. Can you take a look again and make any applicable changes? @carpenterd777 @jlora23

**Comment by carpenterd777** [comment]
> I am unable to load anything past the main page. `peer-types` has changed a bit from 1.0.2 to 1.0.4 and I think that may be why I can't get the details page to load. Can you take a look again and make any applicable changes? @carpenterd777 @jlora23

this is the error:

```
TypeError: undefined is not an object (evaluating 'place.geometry')

This error is located at:
    in DetailedViewScreen (created by MainScreen)
    in MainScreen (created by App)
    in App (created by ExpoRoot)
    in ExpoRoot
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:172:19 in handleException
at node_modules/react-native/Libraries/Core/ReactFiberErrorDialog.js:43:2 in showErrorDialog
at src/hooks/useFetchPlace.ts:25:17 in getPlaceDetails
```

this seems like an error that is related to the server, but I'll take a look

File: __tests__/screens/DetailedViewScreen.test.tsx [hunk 2]
```diff
@@ -2,8 +2,10 @@ import React from "react";
 import { cleanup, render, fireEvent, RenderAPI } from "@testing-library/react-native";
 import axios, { AxiosResponse } from "axios";
 import { useNearbyPlaces } from "../../src/hooks/useNearbyPlaces";
-import { PlaceWithAccesibilityData } from "../../src/util/placeTypes";
-import { PlaceDetailsResponseData } from "@googlemaps/google-maps-services-js";
+import {
+	PlaceDetailsWithAccesibilityData,
+	PlaceWithAccesibilityData,
+} from "../../src/util/placeTypes";
```
Comment (elenirotsides) [inline __tests__/screens/DetailedViewScreen.test.tsx:8]: btw this should be coming from `peer-types`. I think for now you can keep this bc I'm doing the refactor in #294

Comment (carpenterd777) [reply __tests__/screens/DetailedViewScreen.test.tsx:8]: yeah that's why I left it

File: src/screens/SubmitRatingScreen.tsx [hunk 3]
```diff
@@ -21,35 +36,288 @@ export interface SubmitRatingScreenProps {
...
+		return (
+			<View style={styles.yesNoRatingOptions}>
+				<View style={{ flex: 5 }}>
+					<Text style={{ fontSize: 25 }}>{field.renderText}</Text>
+					<PopUp
+						accessibilityLabel={getPopUpProps(
+							field.renderText,
+							"buttonAccessibilityLabel"
+						)}
+						//text={getPopUpProps(field.renderText, "text")}
```
Comment (elenirotsides) [inline src/screens/SubmitRatingScreen.tsx:256]: same thing here

File: src/screens/SubmitRatingScreen.tsx [hunk 4]
```diff
@@ -21,35 +36,288 @@ export interface SubmitRatingScreenProps {
...
+				<CheckBox
+					style={{ flex: 2 }}
+					onClick={() => {
+						if (yesNoCounter[field.fieldName] === 0) {
+							setYesNoCounter({
+								...yesNoCounter,
+								[field.fieldName]: 1,
+							});
+						} else {
+							// yesNoCounter[field.fieldName]  === 1
```
Comment (elenirotsides) [inline src/screens/SubmitRatingScreen.tsx:279]: here as well

File: src/screens/SubmitRatingScreen.tsx [hunk 5]
```diff
@@ -21,35 +36,288 @@ export interface SubmitRatingScreenProps {
...
+						setCounter({
+							...counter,
+							[field.fieldName]: count + INCREMENT_VAL,
+						});
+					}}
+				/>
+			</View>
+		);
+	} else {
+		// field.ratingType === "yes/no"
```
[the remark this answers is not in the document]
Comment (carpenterd777) [reply src/screens/SubmitRatingScreen.tsx:245]: this is a note, not dead code

File: src/screens/SubmitRatingScreen.tsx (hunk 4 above)
Comment (carpenterd777) [reply src/screens/SubmitRatingScreen.tsx:279]: also a note, not dead code

**Pull request merged**